TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -284,5 +329,10 @@ public override DirectoryInfoBase Root
                 return new MockDirectoryInfo(mockFileDataAccessor, mockFileDataAccessor.Directory.GetDirectoryRoot(FullName));
             }
         }
+
+        private void CheckDirectoryExists() 
+        {
+            if (MockFileData == null) throw new FileNotFoundException("File not found", directoryPath);
```

Comment:
Also, those properties throw a FileNotFoundException if the directory does not exist and code tries to set a Time. I'll mimic that too.

```
PS C:\Users\John> $a.LastWriteTime = [System.DateTime]::FromFileTimeUtc(100)
Exception setting "LastWriteTime": "Could not find file 'c:\temperererer'."
At line:1 char:1
+ $a.LastWriteTime = [System.DateTime]::FromFileTimeUtc(100)
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], SetValueInvocationException
    + FullyQualifiedErrorId : ExceptionWhenSetting
```